### PR TITLE
[14.0] [FIX] stock_barcodes: restore read_inventory to old version

### PR DESCRIPTION
**Pull request by renda-dev**
supersedes https://github.com/OCA/stock-logistics-barcode/pull/569

Fixes https://github.com/OCA/stock-logistics-barcode/issues/565 introduced by https://github.com/OCA/stock-logistics-barcode/pull/560

also, added tests that would fail without the latest commit.

### CODE REVIEW

**Comment by francesco-ooops** [comment]
@dalonsod good for you? :)

**Review by dalonsod — APPROVED** [review]

**Comment by dalonsod** [comment]
@alexis-via this solves #565 could you review?

**Comment by francesco-ooops** [comment]
@pedrobaeza @sergio-teruel this solves an issue introduced here: https://github.com/OCA/stock-logistics-barcode/pull/560#issuecomment-1867968119 , could you take a look? thanks!

**Comment by pedrobaeza** [comment]
If others confirmed it's OK, we can merge, but we don't work in this version with this.

File: stock_barcodes/tests/test_stock_barcodes_inventory.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
+from odoo.tests.common import tagged
+
+from .test_stock_barcodes import TestStockBarcodes
+
+
+@tagged("post_install", "-at_install")
+class TestStockBarcodesInventory(TestStockBarcodes):
```
Comment (geomer198) [inline stock_barcodes/tests/test_stock_barcodes_inventory.py:7]: Please add docstring and assertion message.

**Review by geomer198 — COMMENTED** [review]
LGTM!

**Review by geomer198 — APPROVED** [review]

File: stock_barcodes/tests/test_stock_barcodes_inventory.py (hunk 1 above)
Comment (renda-dev) [reply stock_barcodes/tests/test_stock_barcodes_inventory.py:7]: done!

**Comment by francesco-ooops** [comment]
@pedrobaeza please proceed with merge, thanks!

**Pull request merged**